src/Symfony/Component/JsonPath/JsonPathUtils.php

```diff
@@ -85,4 +85,96 @@ public static function findSmallestDeserializableStringAndPath(array $tokens, mi
             'tokens' => $remainingTokens,
         ];
     }
+
+    public static function unescapeString(string $str): string
```

Comment:
just wondering: would there be a way to use json_decode instead of having this method?

Reply:
maybe just for the "u" case?

Reply:
JsonCrawler already tries to use `json_decode` for double-quoted strings before using that utility.

Reply:
> JsonCrawler already tries to use json_decode for double-quoted strings before using that utility.

Maybe I should move this part also in JsonPathUtils so we have the whole unescaping logic at one place instead of trying something with `json_decode` somewhere then try manual unescape